hieroweb/web/src/hiero/web/RpcObjectManager.java

```diff
@@ -0,0 +1,70 @@
+package hiero.web;
+
+import com.google.gson.Gson;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+import java.util.HashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+// Singleton pattern.
```

Comment:
I'm not sure why we need the singleton pattern here. This is going to be a pain to mock and write tests for someday, especially when we have more than one web server talking to the Hiero backend.

There are two other classes that use RpcObjectManager: InitialObject and RpcServer. InitialObject and RpcObjectManager have a circular dependency, which should be fixed (RpcObjectManager uses addObject(initialObject)... and InitialObject.loadTable() later invokes RpcObjectManager.instance.addObject()). The logical order of dependencies looks like RpcServer -> RpcObjectManager -> InitialObject. And I'd suggest just using dependency injection to pass an RpcObjectManager instance to the RpcServer.